### ISSUE TITLE
IPSL organisation

### ISSUE THREAD

**Issue by petitrenaudseb** [itesla/ipsl#84]
When you look at the organization of the IPSL, you can find in the  Electrical/controls/PSSE package all the IEEE regulation.
But these regulations are not only for PSSE but for all the simulation tools

Should we reorganize this package to create a package Electrical/controls/IEEE and let only regulation that you only find in PSSE in the Electrical/controls/PSSE ?

**Comment by MaximeBaudette** [comment]
@petitrenaudseb 
When we started building this library, the idea was to implement models and validate them against a reference software. Thus, we organised the library with sub packages with the name of the reference software. 

I think we should keep it that way for not losing this information (to what reference software it has been validated). Only if in the future we can include some way of determining the validation status, we could reorganize the packages and create one for standard components.

**Comment by petitrenaudseb** [comment]
@MaximeBaudette 
I agree with that, but i hope that the IEEE regulation are the same between PSSE/Eurostag or other software.
If it's the case( i hope), i think there is no need to have IEEE regulations for PSSE and IEEE regulations for Eurostag
If it's not, the actual organisation should stay like that !

**Comment by tinrabuzin** [comment]
I can't think of any examples of the top of my head, but I remember that I saw some differences between IEEE Std. 425-2005 and PSS/E for exciters. I would guess that there are probably some subtle differences in Eurostag as well.

I think @lvanfretti would say that that's the main reason why we are pushing for Modelica.

**Comment by MaximeBaudette** [comment]
@petitrenaudseb 
My main point being, as long as we haven`t come up with a way of flagging the validation against a specific software, we should stick to a package per reference software... even if this means having several "copies" of the same models. 

We did a bit of thinking on our side as to how we should document the validation... but never really did come to a good solution.

**Comment by lvanfretti** [comment]
The way to preserve consistency is that when a model is validated in AAA, it goes under its corresponding AAA classification.

If we want "Generic" IEEE models, that implement the specification according to the standard, BUT not validated against AAA, they should have their own classification as IEEE_Generic.

This should also be the case for IEC/CIM model specifications in the future.

So, we can introduce IEEE_Generic and IECXXX_Generic categories under /Electrical/Controls/, HOWEVER, what has been implemented under /Electrical/Controls/PSSE/ must stay there, because it is not generic - they have been severely "treated" so to match PSS/E. 

By generic we must understand that the behavior or a model is defined by the equations in the standard ONLY, and not have to introduce hacks to match what the implementation in a tool did during their interpretation of the standard.

**Comment by dietmarw** [comment]
[iPSLValidate.mo.zip](https://github.com/itesla/ipsl/files/109527/iPSLValidate.mo.zip)
Just a simple example how you **could** implement this. Problem is that it uses Dymola specific tool annotations which don't work in OMEdit (for now). Other tools might actually support it but I don't have access to SimulationX.

![ipslvalidate](https://cloud.githubusercontent.com/assets/9332/12672531/447b91c0-c677-11e5-96f8-99c5dfff662f.png)

[iPSLValidate.mo.zip](https://github.com/itesla/ipsl/files/109549/iPSLValidate.mo.zip)

**Comment by dietmarw** [comment]
I should perhaps clarify, the **display** and ease of selecting the different options is better in Dymola because of the tool specific annotation `__Dymola_CheckBox`, `__Dymola_compact` and `__Dymola_descriptionLabel`. The model itself is still legal Modelica and would work also in other tools out of the box. Here an example how OMEdit displays it:

![ipslvalitdateom](https://cloud.githubusercontent.com/assets/9332/12673121/0f5ea6b8-c67b-11e5-81f9-fad7a82eb217.png)

As you can see there is also missing support for the standard `Dialog(group)` annotation in the latest development verion. Might be just a regression bug so I've [reported this to OpenModelica](https://trac.openmodelica.org/OpenModelica/ticket/3655)
